File: 2016-maximum-difference-between-increasing-elements/main.go
func maximumDifference(nums []int) int {
    
    if nums == nil {
        return 0
    }

    maxDifference := - 1
    minValue := nums[0] 

    for _, num := range nums {
        if num > minValue {
            maxDifference = max(maxDifference, num - minValue)
        }else {
            minValue = num
        }
    }


    return maxDifference
}

func max(a, b int) int {
    if a < b {
        return b
    }

    return a
}

File: 2016-maximum-difference-between-increasing-elements/main.py
class Solution:
    def maximumDifference(self, nums: List[int]) -> int:
        
        max_difference = -1
        min_value = nums[0]

        for i in range(1, len(nums)):
            if nums[i] > min_value:
                max_difference = max(max_difference, nums[i] - min_value)
            else:
                min_value = nums[i]
                
        return max_difference
            
